docs/configuration-files.md

```diff
@@ -224,6 +226,64 @@ rules:
     severity: ERROR
 ```
 
+### `metavariable-comparison`
+
+The `metavariable-comparison` operator compares metavariables against a basic [Python comparison](https://docs.python.org/3/reference/expressions.html#comparisons)
+expression. This is useful for filtering results based on a [metavariable's](/docs/pattern-features.md#metavariables) value.
+
+**Example**
+
+The `metavariable-comparison` operator is a mapping which requires the
+`metavariable` and `comparison` keys. It can be combined with other pattern operators:
+
+```yaml
+rules:
+  - id: superuser-port
+    patterns:
+      - pattern: set_port($ARG)
+      - metavariable-comparison:
+          metavariable: '$ARG'
+          comparison: '$ARG < 1024'
```

Comment:
same, why the quote? The main interest of yaml is to remove the need for quotes no?

Reply:
Hah, I think YAML provides a bit more than lack of quotes over JSON :D

I've been burned one too many times by YAML interpreting string data as a different scalar type, so I always quote strings by default now /shrug. An interesting example here was including git short hashes in YAML data, so on the off chance your commit hash was like `1234567` YAML would interpret it as an integer instead of the usual string like `5dbfa6d`. That was a fun bug to track down.

I don't have strong feelings - feel free to change the examples in the docs if you'd like.

Reply:
Got it, no the examples are fine, I was just worried I misunderstood something with YAML.